File: frontend/src/services/EmployeeService.ts
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api';

class EmployeeService {
    getAllEmployees() {
        return axios.get(API_BASE_URL);
    }

    getEmployeeById(id: string) {
        return axios.get(`${API_BASE_URL}/${id}`);
    }

    createEmployee(employee: any) {
        return axios.post(API_BASE_URL, employee);
    }

    updateEmployee(id: string, employee: any) {
        return axios.put(`${API_BASE_URL}/${id}`, employee);
    }

    deleteEmployee(id: string) {
        return axios.delete(`${API_BASE_URL}/${id}`);
    }
}

export default new EmployeeService();

File: backend/src/main/java/de/saho/backend/service/EmployeeService.java
package de.saho.backend.service;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import de.saho.backend.exeption.ResourceNotFoundException;
import de.saho.backend.model.Employee;
import de.saho.backend.repo.EmployeeRepo;

import java.util.List;
import java.util.Optional;


@Service
@RequiredArgsConstructor
public class EmployeeService {


    private final EmployeeRepo employeeRepo;

    public List<Employee> getAllEmployees() {
        return employeeRepo.findAll();
    }

    public Employee getEmployeeById(String id) {
        return employeeRepo.findById(id)
                .orElseThrow(() -> new RuntimeException("Not Found"));
    }


    public Employee saveEmployee(Employee employee) {
        return employeeRepo.save(employee);
    }

    public void deleteEmployee(String id) {
        Optional<Employee> employeeOptional = employeeRepo.findById(id);
        if (employeeOptional.isPresent()) {
            employeeRepo.deleteById(id);
        } else {
            throw new ResourceNotFoundException("Employee not found with id: " + id);
        }
    }

    public Employee updateEmployee(String id, Employee employeeDetails) {
        Employee existingEmployee = employeeRepo.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Employee not found with id: " + id));

        existingEmployee.setFirstName(employeeDetails.getFirstName());
        existingEmployee.setLastName(employeeDetails.getLastName());
        existingEmployee.setEmailId(employeeDetails.getEmailId());

        return employeeRepo.save(existingEmployee);
    }






}
